### PR TITLE
support old setuptools on el7

### PR DESCRIPTION
**Pull request by jarodwilson**
The ancient setuptools in el7 behaves differently from more recent
setuptools, which results in unit test failures when running on an el7
host.

A quick illustration of the differences:

  # Python 2.7.13 (fedora)
  >>> from pkg_resources import parse_version
  >>> parse_version('1.0.1')
  <Version('1.0.1')>
  >>> str(parse_version('1.0.1'))
  '1.0.1'

  # Python 2.7.5 (rhel 7)
  >>> from pkg_resources import parse_version
  >>> parse_version('1.0.1')
  ('00000001', '00000000', '00000001', '*final')
  >>> str(parse_version('1.0.1'))
  "('00000001', '00000000', '00000001', '*final')"

With this change, all 12 test failures I was previously encountering are
now passing. The key to this change is standardizing the old setuptools
output to look just like current setuptools output, using a new function
sanitize_version() in osbs/util.py, rather than calling parse_version()
directly.

A special nod to
http://www.programcreek.com/python/example/7930/pkg_resources.parse_version
for Example 4, which provides the basis for this work-around.

Resolves: #564

Signed-off-by: Jarod Wilson <jarod@redhat.com>

### CODE REVIEW

File: osbs/build/build_request.py [hunk 1]
```diff
@@ -26,7 +25,7 @@
 from osbs.build.spec import BuildSpec
 from osbs.constants import SECRETS_PATH, DEFAULT_OUTER_TEMPLATE, DEFAULT_INNER_TEMPLATE, DEFAULT_CUSTOMIZE_CONF
 from osbs.exceptions import OsbsException, OsbsValidationException
-from osbs.utils import looks_like_git_hash, git_repo_humanish_part_from_uri
+from osbs.utils import looks_like_git_hash, git_repo_humanish_part_from_uri, sanitize_version
```
Comment (jarodwilson) [inline osbs/build/build_request.py]: I'm told up to 99 chars is allowed in this project, and looks_like_git_hash was already there, not something I'm adding. It could be removed, but that's really not part of this change.

File: osbs/build/build_request.py [hunk 2]
```diff
@@ -57,7 +56,7 @@ def __init__(self, build_json_store, inner_template=None,
         self._customize_conf = None  # site customize conf for _inner_template
         self._dj = None
         self._resource_limits = None
-        self._openshift_required_version = parse_version('1.0.6')
+        self._openshift_required_version = sanitize_version('1.0.6')
```
Comment (lcarva) [inline osbs/build/build_request.py]: I think we still want the "raw" return value of `parse_version` here. By always sanitizing this value, we lose the main benefit of `parse_version` which is the ability to compare versions. The code right now doesn't actually make use of this value, but it did in the past and will likely use it again in the future.

The only place that we want to sanitize this value is in the `render_orchestrate_build` method because the value is encoded as json. In this case, we need to get back to the raw string value. But anywhere else in the code it should be using whatever `parse_version` returns.

File: tests/test_utils.py [hunk 3]
```diff
@@ -332,3 +333,26 @@ def test_labels(labels, fnc, expect):
...
+    ('bacon', '6.6.6', False, False),
+])
+def test_sanitize_version(v1, v2, match, valid):
+    if valid:
+        if match:
+            assert sanitize_version(parse_version(v1)) == v2
+        else:
+            assert sanitize_version(parse_version(v1)) != v2
+    else:
+        try:
```
Comment (lcarva) [inline tests/test_utils.py]: Why not?
```python
with pytest.raises(TheException):
  sanitize_version(parse_version(v1))
```

File: tests/test_utils.py [hunk 4]
```diff
@@ -332,3 +333,26 @@ def test_labels(labels, fnc, expect):
...
+    ('5.3', '5.3.0', True, True),
+    ('2', '2.0.0', True, True),
+    ('7.3.1', '4.5.9', False, True),
+    ('9', '3.2.1', False, True),
+    ('6.8', '6.8.1', False, True),
+    ('bacon', '6.6.6', False, False),
+])
+def test_sanitize_version(v1, v2, match, valid):
+    if valid:
+        if match:
```
Comment (lcarva) [inline tests/test_utils.py]: Not sure if there's a lot of value in the `match` attribute. This is a simple string comparison.

File: osbs/utils.py [hunk 5]
```diff
@@ -23,6 +23,7 @@
 from datetime import datetime
 from io import BytesIO
 from hashlib import sha256
+from pkg_resources import parse_version
```
Comment (lcarva) [inline osbs/utils.py]: Let's fix this.

File: tests/build_/test_build_request.py [hunk 6]
```diff
@@ -24,7 +24,6 @@
 
 from flexmock import flexmock
 import pytest
-from pkg_resources import parse_version
```
Comment (lcarva) [inline tests/build_/test_build_request.py]: Don't we need this import?

Comment (jarodwilson) [inline tests/build_/test_build_request.py]: No, it's actually currently imported twice, that's just removing the second redundant import.

File: osbs/utils.py (hunk 5 above)
Comment (jarodwilson) [inline osbs/utils.py]: Argh. Stray relic from earlier renditions, thought I'd removed it.

**Comment by twaugh** [comment]
Submitted https://github.com/projectatomic/osbs-client/pull/584 to fix the stickler diagnostic.
